### PR TITLE
Mbkiss benchmarking experiments

### PR DESCRIPTION
**Pull request by mbkiss**

### CODE REVIEW

**Comment by AnderBiguri** [comment]
Should we constrain this to just 2DeteCT? Or should we in principle allow for other datasets?

**Comment by AnderBiguri** [comment]
Also, should `Denoising` be called `Low dose` or something like that?

**Comment by AnderBiguri** [comment]
Also for  consideration: maybe we don't need 6 sparse angle experiments, we can reduce to 4 or so.

**Comment by AnderBiguri** [comment]
Also, I remember us discussing that perhaps some of these cases we wanted the recon done by the loader, rather than using the standard. Do we want this? or was it for the other work? of so, do put `params.data_loader_params.do_recon=True`

**Comment by AnderBiguri** [comment]
All experiments are tested, working template file with documentation for all experiments added.

**Comment by mbkiss** [comment]
> Should we constrain this to just 2DeteCT? Or should we in principle allow for other datasets?

I think for the moment we should keep it restricted to 2DeteCT. It would not work with any other dataset since they would not offer the same modes as 2DeteCT does. If people want to set up similar experiments for other datasets they can get inspired by the file but unifying it in such a way that some (not even all) experiments in there would also work for other datasets is not worth it imho...

Also, there is the ct_experiments.py which we wanted to keep general. So that would be my recommended approach :)

**Comment by mbkiss** [comment]
> Also, should `Denoising` be called `Low_dose` or something like that? to avoid confusion with a `recon2recon` task.

Yes, I can see how: "LowDoseCTReconstruction" would fit the other experiment classes well :) Please do so :)

**Comment by mbkiss** [comment]
> Also for consideration: maybe we don't need 6 sparse angle experiments, we can reduce to 4 or so.

I fully agree with it. We want to limit the number of experiments that we have to run, because we would need to run each additional experiment with 12 algorithms and that will cost a lot of computation time. 

I currently have that many, because I am unsure at the moment when significant artifacts will show it the classical FBP recons in these subsampled situations. If there is nothing showing until 180 we can drop the initial ones. If 60 is way too severe than we can get rid of that too.

My recommendation for the way forward would be to leave them in and then test them with one of the more stable and known algorithms to see what the performance is like for these boundary cases and then decide which to keep. Do you agree?

**Comment by mbkiss** [comment]
> Also, I remember us discussing that perhaps some of these cases we wanted the recon done by the loader, rather than using the standard. Do we want this? or was it for the other work? of so, do put `params.data_loader_params.do_recon=True`

So, we referred to methods where the input into the ML algorithm would be e.g. a FBP-recon of a sinogram. But that would be part of the algorithm would it not? Because we still want to load a specific sinogram(-subsampling) as original input and an already well defined iterative reconstruction as a target, wouldn't we?

File: LION/experiments/ct_benchmarking_experiments.py [hunk 1]
```diff
@@ -142,12 +108,11 @@ def default_parameters(dataset="2DeteCT"):
...
         param.data_loader_params.target_mode = "mode2"
-        
+
         return param
 
 
 class LimitedAngle90CTRecon(Experiment):
-
     def __init__(self, experiment_params=None, dataset="2DeteCT", datafolder=None):
         super().__init__(experiment_params, dataset, datafolder)
```
Comment (mbkiss) [inline LION/experiments/ct_benchmarking_experiments.py:118]: Below there should then also be 
param.name = (
            "90 degrees Limited Angle CT reconstruction experiment from mode 2"
        )

File: LION/experiments/ct_benchmarking_experiments.py [hunk 2]
```diff
@@ -171,12 +136,11 @@ def default_parameters(dataset="2DeteCT"):
...
         param.data_loader_params.target_mode = "mode2"
-        
+
         return param
 
 
 class LimitedAngle60CTRecon(Experiment):
-
     def __init__(self, experiment_params=None, dataset="2DeteCT", datafolder=None):
         super().__init__(experiment_params, dataset, datafolder)
```
Comment (mbkiss) [inline LION/experiments/ct_benchmarking_experiments.py:146]: Below there should then also be 
param.name = (
            "60 degrees Limited Angle CT reconstruction experiment from mode 2"
        )

**Comment by mbkiss** [comment]
> All experiments are tested, working template file with documentation for all experiments added.

Fantastic! I have read through them and only miss two things to make your edits consistent which I added as reviews... but other than that I am very happy with the content!

Shall we make the last experiment then also: "BeamHardeningCTReconstruction"?

File: LION/experiments/ct_benchmarking_experiments.py (hunk 1 above)
Comment (AnderBiguri) [reply LION/experiments/ct_benchmarking_experiments.py:118]: It is! Github may have hiden you unchanged code, but its there 5 lines below.

Comment (mbkiss) [reply LION/experiments/ct_benchmarking_experiments.py:118]: I know that the basic content is there. But you changed the style of the code with brackets and new lines, which should be the same for all experiments, shouldn't it?

Comment (AnderBiguri) [reply LION/experiments/ct_benchmarking_experiments.py:118]: @mbkiss thats the `black` autoformater, we let it do what it does 🤷
